File: studyUrl/src/main/java/com/surl/studyurl/domain/member/repository/MemberRepository.java
package com.surl.studyurl.domain.member.repository;

import com.surl.studyurl.domain.member.entity.Member;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member,Long> {
    Optional<Member> findByUserid(String userid);
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/domain/member/repository/MemberRepository.kt
package com.krstudy.kapi.domain.member.repository

import com.krstudy.kapi.domain.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface MemberRepository : JpaRepository<Member, Long> {
    fun findByUsername(username: String): Member?
    fun findByUserid(userId: String): Member?
    fun findByJwtToken(jwtToken: String): Member? // JWT 토큰으로 멤버 조회
    fun findByUsernameContaining(username: String): List<Member>  // 키워드로 사용자 검색
    fun findByUsernameAndUserEmailAndAccountType(username: String, userEmail: String, accountType: String = "WEB"): Optional<Member>
}
